### PR TITLE
Change menu bar to custom version

### PR DESCRIPTION
**Pull request by Happyrobot33**
not sure how to fix the extra commits. lets see if it works this time

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -520,9 +520,114 @@
...
+				<svg preserveAspectRatio="none" viewBox="0 0 24 24">
+					<use href="#icon_min_max">
+				</svg>
+			</button>
+			<button id="closeButton">
+				<svg preserveAspectRatio="none" viewBox="0 0 24 24">
+					<use href="#icon_close_program">
+				</svg>
+			</button>
+		  </div>
```
Comment (noahkra) [inline index.html:629]: Fix indentation please

File: index.html [hunk 2]
```diff
@@ -563,13 +668,14 @@
...
-			</div>
-			<div id="versionNumber" style="position: fixed; text-align: center; right: 5px; bottom: 0px; color: var(--text-main-L-col); display: inline-block">0.0.0</div>
+			<div id="settingsInfo">Nothing here at the moment</div>
+			<div id="versionNumber">0.0.0</div>
 		</div>
 		<svg>
 			<defs>
+				<path id="icon_min_max" d="M18 4H6c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H6V6h12v12z"></path>
+				<path id="icon_min" d="M19 13H5v-2h14v2z"></path>
+				<path id="icon_close_program" d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"></path>
```
Comment (noahkra) [inline index.html:678]: add title to id?

File: index.js [hunk 3]
```diff
@@ -13,20 +13,21 @@ const analyser                         = require('./analyser.js');
...
-const path                             = require('path');
-const fs                               = require('fs');
+const winscreen											= require('electron').remote.screen;
+const xml2js											= require('xml2js').parseString;
+const { dialog, getCurrentWindow}						= require('electron').remote;
+const { shell, ipcRenderer, webFrame }					= require('electron');
+const request											= require('request');
+const semVer											= require('semver');
+const path												= require('path');
+const fs												= require('fs');
```
Comment (noahkra) [inline index.js:23]: why are the "=" not aligned and why are they so far out? Align them all one space after `const { shell, ipcRenderer, webFrame }` please.

File: index.js [hunk 4]
```diff
@@ -13,20 +13,21 @@ const analyser                         = require('./analyser.js');
...
 
 const saveLocation = process.env.APPDATA + "\\splitsAnalyser\\";
 
 // Version
 const version = "0.4.1";
 d3.select("#versionNumber").text(version); //set the version text to the correct number
+d3.select("#titleVersionNumber").text(version); //set the version text to the correct number
```
Comment (noahkra) [inline index.js:30]: unnecessary. see 3 lines below.

File: index.js [hunk 5]
```diff
@@ -170,6 +171,25 @@ window.addEventListener('keyup', (event) => {
...
+	}
+	else{
+		getCurrentWindow().maximize();
+	}
+});
+// This handles the Close button
+document.getElementById("closeButton").addEventListener('click', () =>{
+	getCurrentWindow().close();
+});
+
```
Comment (noahkra) [inline index.js:192]: Still not using d3...

File: index.html [hunk 6]
```diff
@@ -520,9 +520,114 @@
...
+				-webkit-app-region: no-drag;
+			}
+
 		</style>
 	</head>
 	<body>
+		<div id="titleBar"> <!-- use to define the height of the status bar -->
+		  <div id="titleVersionNumber">0.0.0</div>
+		  <div id="titleBarAppName"> <!-- use to define the location of the status bar title -->
+			<div><p>Splits Analyser</p></div>
```
Comment (noahkra) [inline index.html:611]: Just use document.title for this as that's already defined in index.js

File: index.html [hunk 7]
```diff
@@ -520,9 +520,114 @@
...
+				background: transparent; 
+				border: 0px;
+				width: 30px;
+				height: 30px;
+				fill: var(--item-col)
+			}
+
+			#minimizeButton:hover {
+				background: var(--item-col);
+				fill: var(--highlight-col)
```
Comment (noahkra) [inline index.html:533]: Add transition like with the other buttons. same goes for the other :hover's you added

**Comment by Happyrobot33** [comment]
done

File: index.html [hunk 8]
```diff
@@ -520,9 +520,116 @@
...
+				right: 0%;
+				-webkit-app-region: no-drag;
+			}
+
 		</style>
 	</head>
 	<body>
+		<div id="titleBar"> <!-- use to define the height of the status bar -->
+			<div id="titleBarContainer"> <!-- use to define the location of the status bar title -->
+			<div id="titleBarName">SplitsAnalyser</div>
```
Comment (noahkra) [inline index.html:613]:
```suggestion
				<div id="titleBarName"/>
```
Name is already handled within index.js, so unnecessary to add here again.

File: index.js [hunk 9]
```diff
@@ -170,6 +171,27 @@ window.addEventListener('keyup', (event) => {
...
 
+// New Status bar events
+// This handles the minimize button
+d3.select("#minimizeButton").on("click", function() {
+	getCurrentWindow().minimize();
+})
+
+// This handles the Min Max button
+d3.select("#minMaxButton").on("click", function() {
+	if(getCurrentWindow().isMaximized()){
```
Comment (noahkra) [inline index.js:182]:
```suggestion
	if (getCurrentWindow().isMaximized()) {
```

File: index.js [hunk 10]
```diff
@@ -170,6 +171,27 @@ window.addEventListener('keyup', (event) => {
...
+		getCurrentWindow().unmaximize();
+	}
+	else{
```
Comment (noahkra) [inline index.js:185]:
```suggestion
	} else {
```

File: index.html [hunk 11]
```diff
@@ -520,9 +520,116 @@
...
+			<button id="minMaxButton">
+				<svg preserveAspectRatio="none" viewBox="0 0 24 24">
+					<use href="#icon_controls_min_max">
+				</svg>
+			</button>
+			<button id="closeButton">
+				<svg preserveAspectRatio="none" viewBox="0 0 24 24">
+					<use href="#icon_controls_close_program">
+				</svg>
+			</button>
```
Comment (noahkra) [inline index.html:630]:
```suggestion
				<button id="minimizeButton">
					<svg preserveAspectRatio="none" viewBox="0 0 24 24">
						<use href="#icon_controls_min">
					</svg>
				</button>
				<button id="minMaxButton">
					<svg preserveAspectRatio="none" viewBox="0 0 24 24">
						<use href="#icon_controls_min_max">
					</svg>
				</button>
				<button id="closeButton">
					<svg preserveAspectRatio="none" viewBox="0 0 24 24">
						<use href="#icon_controls_close_program">
					</svg>
				</button>
```

File: index.html (hunk 8 above)
Comment (Happyrobot33) [reply index.html:613]: thats there incase it doesnt work

**Pull request merged**